### PR TITLE
Lesson 10: Jobs API Part 2

### PR DESCRIPTION
**Pull request by catbus00**

### CODE REVIEW

**Comment by catbus00** [comment]
Deployed on Render here: ( https://jobs-api-jillian.onrender.com)

File: controllers/auth.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  res.status(StatusCodes.CREATED).json({ user: { name: user.name }, token });
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    throw new BadRequestError('Please provide email and password');
+  }
+  const user = await User.findOne({ email });
```
Comment (jcheng1022) [inline controllers/auth.js:17]: same issue as previous PR, the `if(!user).... ` code should come directly after the check for user. small fix

File: middleware/error-handler.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
+    )} field, please choose another value`;
+    customError.statusCode = 400;
+  }
+  if (err.name === 'CastError') {
+    customError.msg = `No item found with id: ${err.value}`;
+    customError.statusCode = 404;
   }
-  return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ err })
-}
+  // return res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ err });
```
Comment (jcheng1022) [inline middleware/error-handler.js]: really small detail but be sire to clean up commented code!

**Comment by catbus00** [comment]
Thank you so much @jcheng1022 ! I really appreciate your feedback and taking the time to review. 

There were a few other pull requests that have not been reviewed yet. Would it be possible if you could take a look if you have the chance, please? :) https://github.com/catbus00/node-express-course/pulls

**Pull request merged**